### ISSUE TITLE
Missing index entries for definitions of the terms "satisfies" and "models"

### ISSUE THREAD

**Issue by Quuxplusone** [cplusplus/draft#3523]
The draft repeatedly uses the terms of art "satisfies/satisfy" and "models/model" without pointing to their definitions, which would be fine, except that these terms of art *also* do not appear in the Index!

It's hard to search for "model" in the full text because of "memory model" and "object model", but I have at least tracked it down to a non-normative example http://eel.is/c++draft/concepts.equality#8 —
> `T` fails to meet the implicit requirements of `C`, so `T` satisfies but does not model `C`. Since implementations are not required to validate the syntax of implicit requirements, it is unspecified whether an implementation diagnoses as ill-formed a program that requires `C<T>`.

http://eel.is/c++draft/temp.constr.atomic#3 seems to provide a normative definition for _satisfy_.
> To determine if an atomic constraint is _satisfied_, the parameter mapping and template arguments are first substituted into its expression. If substitution results in an invalid type or expression, the constraint is not satisfied. Otherwise, the lvalue-to-rvalue conversion is performed if necessary, and `E` shall be a constant expression of type `bool`. The constraint is satisfied if and only if evaluation of `E` results in `true`.

I have not discovered the normative definition for _model_.

**Comment by jensmaurer** [comment]
"satisfies" is fine and covered by the core language. It seems that "models" is only defined with specific concepts. We frequently say "a type T models C only if ..." I could not find a generic definition of "models" in the library front matter (where it would belong).

**Comment by jwakely** [comment]
Right, we only use "models" in the context of specific concepts, and the definition of each concept says what it means to model that concept.

@CaseyCarter is already on the hook to write a paper clarifying these terms, to resolve some open LWG issues and NB comments.

**Comment by CaseyCarter** [comment]
> Right, we only use "models" in the context of specific concepts, and the definition of each concept says what it means to model that concept.
> 
> @CaseyCarter is already on the hook to write a paper clarifying these terms, to resolve some open LWG issues and NB comments.

Yes, feel free to assign (the remainder not covered by Jens's PR) this issue to me.
